### ISSUE TITLE
[Question]: Ernie-layout support for multi-answers VQA finetuning

### ISSUE THREAD

**Issue by stefanqxb** [PaddlePaddle/PaddleNLP#3941]
### 请提出你的问题

Hi there, 
I have another question here, regrading the VQA dataset, seems now for each question, we only have one answer, for example: {"text": xxx, "answer_start": [2], "answer_end":[5]}, which the answer is a consective span from the text, so Im wondering if we support questions with answers which is not consective, e.g. multi-span VQA ??

For example, given a OCR result is ["This", "is", "a", "dummy", "test", "with", "right", "answer", "A1", "and", "A1", "has"....(some more words here)..,"another", "right", "answer", "A2",.....], so in this case, for the question: `What are the right answers` ?, the answer should be `A1` and `A2`, however, `A1` and `A2` are not next to each other, then how should we specify these kinda answer for the training ? Thanks for the help

**Comment by linjieccc** [comment]
@stefanqxb Hi,

Thanks for your attention on ernie-layout.

CrossEntropyLoss and softmax are used on the task of fine-tuning ernie-layout on docvqa_zh dataset. 

For multi answers VQA scenario, I think you should replace CrossEntropyLoss by BCELoss in training and replace softmax by sigmoid in the computation of start and end probabilities of spans.

**Comment by stefanqxb** [comment]
Hi @linjieccc 
    Thanks for the answers, so I have two questions here now :D 
    1. do we have some instructions where those codes are so I could replace as u suggested ?
    2. How should  we formulate our training dataset for this mutli-answers cases?  Given that, now the **answer** part in the training data is set is `{"text": ["xxxx"], "answer_start": [1], "answer_end":[4]}` which is for the single-answer secnario, so should I change that into like `{"text",["A", "B", "C"], "answer_start":[1, 5, 7], "answer_end": [3, 6, 8]}` for the multi-ones where [1,3] is for A ,  [5,6] for B, [7-8] for C ? Thanks for the help here

**Comment by stefanqxb** [comment]
Hi @linjieccc ,
     I also got another question now about the data format, as when I checked the `docvqa_zh` , I see that the text is split into **char-level** and with the corresponding bbox,  this is feed for training,  so wondering how does the paddle get this **char-level** text and its bbox, is there any tool to achieve the same for our finetune data ? Thanks for th help.

**Comment by linjieccc** [comment]
@stefanqxb Hi,

ErnieLayoutForQuestionAnswering:
https://github.com/PaddlePaddle/PaddleNLP/blob/988204dfbf8e8c1c0e733b25996a9d47ee48ffa4/paddlenlp/transformers/ernie_layout/modeling.py#L1184-L1265

postprocess for VQA task:
https://github.com/PaddlePaddle/PaddleNLP/blob/988204dfbf8e8c1c0e733b25996a9d47ee48ffa4/model_zoo/ernie-layout/utils.py#L1029-L1061

You can also refer to the decoding method of UIE, which is for multi-answer scenario.

**Comment by linjieccc** [comment]
> Hi @linjieccc , I also got another question now about the data format, as when I checked the `docvqa_zh` , I see that the text is split into **char-level** and with the corresponding bbox, this is feed for training, so wondering how does the paddle get this **char-level** text and its bbox, is there any tool to achieve the same for our finetune data ? Thanks for th help.

We got the **char-level** bbox by calculating the char width for each text segment

**Comment by stefanqxb** [comment]
Hi @linjieccc 
    Thank you so much for the detailed explanation, I can start looked into the corresponding scripts as u suggested above.
And as for the `char-level` bbox, does paddle has some built-in tool for that calculations or do I have to build on ourselves ?

**Comment by stefanqxb** [comment]
Hi @linjieccc:
     any updates for the above question?

**Comment by linjieccc** [comment]
@stefanqxb Hi, we got the char-level box by dividing the width of the segment-level box by the number of characters, please refer to the code here https://github.com/PaddlePaddle/PaddleNLP/blob/develop/paddlenlp/utils/tools.py#L330
